src/rules/selector-class-pattern/index.js

```diff
@@ -13,23 +14,49 @@ export const messages = ruleMessages(ruleName, {
   expected: selectorValue => `Expected class selector ".${selectorValue}" to match specified pattern`,
 })
 
-export default function (pattern) {
+export default function (pattern, options) {
   return (root, result) => {
     const validOptions = validateOptions(result, ruleName, {
       actual: pattern,
       possible: [ isRegExp, isString ],
+    }, {
+      actual: options,
+      possible: {
+        resolveNestedSelectors: isBoolean,
+      },
+      optional: true,
     })
     if (!validOptions) { return }
 
+    // In context of a nested selector, some combinations
+    // are tested mutliple times, to avoid triggering multiple
+    // identical errors, we will use an internal list of checked rules
+    var checked = []
+
     const normalizedPattern = isString(pattern) ? new RegExp(pattern) : pattern
 
     root.walkRules(rule => {
       if (cssRuleHasSelectorEndingWithColon(rule)) { return }
-      selectorParser(checkSelector).process(rule.selector)
+
+      if (options && options.resolveNestedSelectors) {
+        resolveNestedSelector(rule.selector, rule).forEach(selector => {
```

Comment:
Have you considered the suggestion in the comment above --- only resolving selectors if the selector contains a `&` that is part of an identifier (i.e. not its own selector, like `& > div`, but part of another, like `&-foo`)?